app/reducers/comments.js

```diff
@@ -15,62 +16,45 @@ export type CommentEntity = {
   author: UserEntity | null
 };
 
+type State = any;
 /**
  * Used by the individual entity reducers
  */
 export function mutateComments(forTargetType: string) {
-  return (state: any, action: any) => {
-    switch (action.type) {
-      case Comment.ADD.SUCCESS: {
-        const [serverTargetType, targetId] = action.meta.commentTarget.split(
-          '-'
-        );
-        const targetType = getEntityType(serverTargetType);
-        if (targetType !== forTargetType) {
-          return state;
-        }
-
-        return {
-          ...state,
-          byId: {
-            ...state.byId,
-            [targetId]: {
-              ...state.byId[targetId],
-              comments: [
-                ...state.byId[targetId].comments,
-                action.payload.result
-              ]
+  return (state: State, action: any) => {
+    return produce(
+      state,
+      (newState: State): void => {
+        switch (action.type) {
+          case Comment.ADD.SUCCESS: {
+            const [
+              serverTargetType,
+              targetId
+            ] = action.meta.commentTarget.split('-');
+            const targetType = getEntityType(serverTargetType);
+            if (targetType === forTargetType) {
+              newState.byId[targetId].comments.push(action.payload.result);
             }
           }
-        };
-      }
-
-      default: {
-        return state;
+        }
       }
-    }
+    );
   };
 }
 
-function mutate(state: any, action: any) {
-  switch (action.type) {
-    case Comment.DELETE.SUCCESS: {
-      return {
-        ...state,
-        byId: {
-          ...state.byId,
-          [action.meta.id]: {
-            ...state.byId[action.meta.id],
-            text: null,
-            author: null
-          }
-        }
-      };
-    }
-    default: {
-      return state;
+type CommentState = any;
+
+function mutate(state: CommentState, action: any): CommentState {
+  return produce(
```

Comment:
Maybe `const mutate..... => produce()` ? I know you didn't write it, but when you're here, refactoring, we might as well make it nice (Like you did with mutateArticle)